src/registry.rs

```diff
@@ -668,6 +681,100 @@ mod deserializers {
     }
   }
 
+  struct OptionVisitor<T> {
+    marker: std::marker::PhantomData<fn() -> Option<T>>,
+  }
+
+  impl<'de, T> Visitor<'de> for OptionVisitor<T>
+  where
+    T: DeserializeOwned,
+  {
+    type Value = Option<T>;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+      formatter.write_str("Option wrapping string")
+    }
+
+    fn visit_none<E>(self) -> Result<Self::Value, E>
+    where
+      E: de::Error,
+    {
+      Ok(None)
+    }
+
+    fn visit_map<M>(self, mut map: M) -> Result<Self::Value, M::Error>
+    where
+      M: MapAccess<'de>,
+    {
+      while map
+        .next_entry::<de::IgnoredAny, de::IgnoredAny>()?
+        .is_some()
+      {}
+      Ok(None)
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+      A: SeqAccess<'de>,
+    {
+      while seq.next_element::<de::IgnoredAny>()?.is_some() {}
+      Ok(None)
+    }
+
+    fn visit_some<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+    where
+      D: Deserializer<'de>,
+    {
+      let value = T::deserialize(deserializer);
+      match value {
+        Ok(string_value) => Ok(Some(string_value)),
```

Comment:
Seems like this struct should be called some thing like `OptionalStringVisitor`?